build.py

```diff
@@ -23,18 +27,263 @@ def wait_until_container_is_ready():
+        +'      dnsPolicy: ClusterFirst\n' \
+        +'      restartPolicy: Always\n' \
+        +'      terminationGracePeriodSeconds: 30\n' \
+        +'    paramters:\n' \
+        +'    - name: "openshifturl"\n' \
+        +'      displayName: "OpenShift URL"\n' \
+        +'      description: "The OpenShift Master URL - because openshift pods don\'t know their master"\n' \
+        +'      required: true"\n' \
+        +'      value: ' + openshift_url + '\n' \
+        +'  test: False\n' \
+        +'  triggers:\n' \
+        +'  - type: ConfigChange\n'
+    return dc
+
+def oc_create_dc(openshift_url,microserver_url,project,docker_image):
+    proc=subprocess.Popen(['oc','create','-f','-'],stdout=subprocess.PIPE,stderr=subprocess.STDOUT,stdin=subprocess.PIPE)
+    proc.communicate(get_dc_def(openshift_url,microserver_url,project,docker_image).encode())
+    return 1
+
+def oc_service_exists(project,service):
+    result=subprocess.run(['oc', r'-n',project, 'get', 'service', service],stdout=subprocess.PIPE,stderr=subprocess.STDOUT)
+    if(compare_results(result,r'^'+service)):
+        return True
+    return False
+
+#apiVersion: v1
+#kind: Service
+#metadata:
+#  creationTimestamp: 2019-09-26T13:47:48Z
+#  labels:
+#    app: acct-mgt
+#  name: acct-mgt
+#  namespace: acct-mgt
+#  resourceVersion: "879306"
+#  selfLink: /api/v1/namespaces/acct-mgt/services/acct-mgt
+#  uid: 39b978e4-e064-11e9-9cfd-fa163e2bb38b
+#spec:
+#  clusterIP: 172.30.195.49
+#  externalIPs:
+#  - 172.29.230.64
+#  externalTrafficPolicy: Cluster
+#  ports:
+#  - nodePort: 30002
+#    port: 8080
+#    protocol: TCP
+#    targetPort: 8080
+#  selector:
+#    app: acct-mgt
+#    deploymentconfig: acct-mgt
+#  sessionAffinity: None
+#  type: LoadBalancer
+#status:
+#  loadBalancer:
+#    ingress:
+#    - ip: 172.29.230.64
+
+
+def get_svc_def(project,service,port=8080):
+    svc=""
+    if(project is not None and len(project)>0 and port is not None):
+        svc='{ "kind": "Service",' \
+            + '"apiVersion": "v1",' \
+            + '"metadata":' \
+            +  '{ "name": "'+service+'",' \
+            +    '"namespace":"'+project+'" },' \
+            +    '"spec": { ' \
+            +      '"ports": [ ' \
+            +          '{ "protocol": "TCP", "port": '+str(port)+', "targetPort": '+str(port)+' }' \
+            +      '], '\
+            +    '"type":"LoadBalancer",' \
+            +    '"externalTrafficPolicy": "Cluster",' \
+            +    '"selector":{'\
+            +        '"app":"'+project+'",'\
+            +        '"deploymentconfig": "'+project+'" }' \
+            +   '} }'
+    return svc
+
+def oc_create_service(project,port=8080):
+    proc=subprocess.Popen(['oc','-n',project,'create','-f','-'],stdout=subprocess.PIPE,stderr=subprocess.STDOUT,stdin=subprocess.PIPE)
+    json_str=get_svc_def(project,port)
+    print("service json: "+json_str)
+    proc.communicate(json_str.encode())
+    #printing stdout doesn't seem to work.
```

Comment:
It doesn't like look like you wait for the process to terminate here, which means you never have the chance to check for an error. If you're going to call `subprocess.Popen` like this, then you need to call `proc.wait()` and wait for it to terminate before returning to the caller...and then do something if the return code indicates an error.